src/Lecture2.hs

```diff
@@ -79,7 +78,9 @@ return the removed element.
 (Nothing,[1,2,3,4,5])
 -}
 removeAt :: Int -> [a] -> (Maybe a, [a])
-removeAt i l = go before after $ max i (-1) -- detect negative numbers
+removeAt i l
+  | i < 0  =  go before after (-1 :: Int) -- detect negative numbers
```

Comment:
It looks like you can return the answer here immediately and you don't need to handle negative cases in the recursive `go` function 🙂 
```suggestion
  | i < 0  =  (Nothing, l)
```

Reply:
whoa :exploding_head: 

fixed thanks! <3